bot/exts/utils/extensions.py

```diff
@@ -0,0 +1,265 @@
+        r"""
+        Unload currently loaded extensions given their fully qualified or unqualified names.
+
+        If '\*' or '\*\*' is given as the name, all loaded extensions will be unloaded.
+        """  # noqa: W605
+        if not extensions:
+            await ctx.send_help(ctx.command)
+            return
+
+        blacklisted = "\n".join(UNLOAD_BLACKLIST & set(extensions))
+
+        if blacklisted:
+            msg = f":x: The following extension(s) may not be unloaded:```{blacklisted}```"
+        else:
+            if "*" in extensions or "**" in extensions:
+                extensions = set(self.bot.extensions.keys()) - UNLOAD_BLACKLIST
+
+            msg = self.batch_manage(Action.UNLOAD, *extensions)
+
+        await ctx.send(msg)
+
+    @extensions_group.command(name="reload", aliases=("r",), root_aliases=("reload",))
+    async def reload_command(self, ctx: Context, *extensions: Extension) -> None:
+        r"""
+        Reload extensions given their fully qualified or unqualified names.
+
+        If an extension fails to be reloaded, it will be rolled-back to the prior working state.
+
+        If '\*' is given as the name, all currently loaded extensions will be reloaded.
+        If '\*\*' is given as the name, all extensions, including unloaded ones, will be reloaded.
+        """  # noqa: W605
+        if not extensions:
+            await ctx.send_help(ctx.command)
+            return
+
+        if "**" in extensions:
+            extensions = EXTENSIONS
+        elif "*" in extensions:
+            extensions = set(self.bot.extensions.keys()) | set(extensions)
+            extensions.remove("*")
+
+        msg = self.batch_manage(Action.RELOAD, *extensions)
+
+        await ctx.send(msg)
+
+    @extensions_group.command(name="list", aliases=("all",))
+    async def list_command(self, ctx: Context) -> None:
+        """
+        Get a list of all extensions, including their loaded status.
+
+        Grey indicates that the extension is unloaded.
+        Green indicates that the extension is currently loaded.
+        """
+        embed = Embed(colour=Colour.blurple())
+        embed.set_author(
+            name="Extensions List",
+            url=URLs.github_bot_repo,
+            icon_url=URLs.bot_avatar
+        )
+
+        lines = []
+        categories = self.group_extension_statuses()
+        for category, extensions in sorted(categories.items()):
+            # Treat each category as a single line by concatenating everything.
+            # This ensures the paginator will not cut off a page in the middle of a category.
+            category = category.replace("_", " ").title()
+            extensions = "\n".join(sorted(extensions))
+            lines.append(f"**{category}**\n{extensions}\n")
+
+        log.debug(f"{ctx.author} requested a list of all cogs. Returning a paginated list.")
+        await LinePaginator.paginate(lines, ctx, embed, max_size=700, empty=False)
```

Comment:
The `max_size` needs to be increased to 1200, or else you'll run into something like this:
```py
09/21/20 16:06:38 - root DEBUG: Error Encountered: RuntimeError - Line exceeds maximum page size 698, Command: extensions list, Author: f1re#3996, Channel: python-general
09/21/20 16:06:38 - bot.exts.evergreen.error_handler ERROR: Unhandled command error: Line exceeds maximum page size 698
Traceback (most recent call last):
  File "/home/dpham/.local/share/virtualenvs/seasonalbot-oSM7Qold/lib/python3.8/site-packages/discord/ext/commands/core.py", line 85, in wrapped
    ret = await coro(*args, **kwargs)
  File "/home/dpham/code/pydis/seasonalbot/bot/exts/utils/extensions.py", line 171, in list_command
    await LinePaginator.paginate(lines, ctx, embed, max_size=700, empty=False)
  File "/home/dpham/code/pydis/seasonalbot/bot/utils/pagination.py", line 139, in paginate
    paginator.add_line(line, empty=empty)
  File "/home/dpham/code/pydis/seasonalbot/bot/utils/pagination.py", line 59, in add_line
    raise RuntimeError('Line exceeds maximum page size %s' % (self.max_size - len(self.prefix) - 2))
RuntimeError: Line exceeds maximum page size 698
```
```suggestion
        await LinePaginator.paginate(lines, ctx, embed, max_size=1200, empty=False)
```